scripts/migrations/2024_10_remove_duplicate_products_on_wrong_flavors.pl

```diff
@@ -0,0 +1,163 @@
+
+TXT
+	;
+
+my $csv_file;
+
+GetOptions("flavor-code-csv=s" => \$csv_file,);
+
+if (not defined $csv_file) {
+	print STDERR $usage;
+	exit;
+}
+
+open(my $log, ">>", "$data_root/logs/remove_duplicate_products_on_wrong_flavors.log");
+print $log "move_ean8_products_to_new_path.pl started at " . localtime() . "\n";
+
+my $products_collection = get_products_collection();
+my $obsolete_products_collection = get_products_collection({obsolete => 1});
+
+sub move_code_to_other_flavors_codes($code) {
+
+	my $product_id = product_id_for_owner(undef, $code);
+	my $dir = product_path_from_id($product_id);
+
+	my $target_dir = $dir;
+	$target_dir =~ s/[^0-9]//g;
+
+	if (move("$data_root/products/$dir", "$data_root/products/other-flavors-codes/$target_dir")) {
+		print STDERR "moved other flavors code $dir to $data_root/products/other-flavors-codes/$target_dir\n";
+		print $log "moved other flavors code $dir to $data_root/products/other-flavors-codes/$target_dir\n";
+	}
+	else {
+		print STDERR "could not move other flavors code $dir to $data_root/products/other-flavors-codes/$target_dir\n";
+		print $log "could not move other flavors code $dir to $data_root/products/other-flavors-codes/$target_dir\n";
+	}
+	# Delete from mongodb
+	my $id = $code;
+	$products_collection->delete_one({_id => $id});
+	$obsolete_products_collection->delete_one({_id => $id});
+
+	# Also move the image dir if it exists
+	if (-e "$www_root/images/products/$dir") {
+		if (move("$www_root/images/products/$dir", "$www_root/images/products/other-flavors-codes/$target_dir")) {
+			print STDERR
+				"moved other flavors code $dir images to $www_root/images/products/other-flavors-codes/$target_dir\n";
+			print $log
+				"moved other flavors code $dir images to $www_root/images/products/other-flavors-codes/$target_dir\n";
+		}
+		else {
+			print STDERR
+				"could not move other flavors code $dir images to $www_root/images/products/other-flavors-codes/$target_dir\n";
+			print $log
+				"could not move other flavors code $dir images to $www_root/images/products/other-flavors-codes/$target_dir\n";
+		}
+	}
+
+	return;
+}
+
+ensure_dir_created_or_die("$data_root/products/other-flavors-codes");
+ensure_dir_created_or_die("$www_root/images/products/other-flavors-codes");
+
+# Open CSV file
+open(my $csv_fh, "<", $csv_file) or die "Could not open file $csv_file: $!";
+
+while (my $line = <$csv_fh>) {
+	chomp($line);
+	my ($kept_flavor, $code) = split(/\t/, $line);
+	$code = normalize_code($code);
+
+	# Code not numeric? may be header line, skip
+	if ($code !~ /^\d+$/) {
+		next;
+	}
+
+	# Undefined flavor, do nothing
+	if ((not defined $kept_flavor) or ($kept_flavor eq "")) {
+		next;
+	}
+
+	# Check if the product exists on the current flavor
+	my $product_ref = retrieve_product(product_id_for_owner(undef, $code), "include_deleted");
+	if (not defined $product_ref) {
+		print STDERR "code $code does not exist on the current flavor\n";
+		next;
+	}
```

Comment:
Is it worth an error ?
Of course a lot of "off" products won't be on opf. Or do you filtre the CSV to products that should be on this flavor ?

Reply:
I was just going to use the CSV as is and run the script on all flavors.